src/Views/Form/Templates/Sequoia/views/subscription-details.php

```diff
@@ -0,0 +1,101 @@
+<?php
+
+use Give\Addon\Recurring as RecurringAddon;
+use Give\Donation\Donation;
+use function Give\Helpers\Form\Template\Utils\Frontend\getPaymentId;
+
+$donationID = getPaymentId();
+$donation   = new Donation( $donationID );
+
+// Exit if donation is not subscription.
+if ( ! RecurringAddon::isActive() || ! $donation->isRecurring() ) {
+	return '';
+}
+
+$subscriptionDB = new Give_Subscriptions_DB();
```

Comment:
@ravinderk @JasonTheAdams I can see every form template we creating needing to support recurring donations. How can this be more reusable across multiple templates? Can we have a common library of shared components?

Reply:
@DevinWalker It is really difficult to build a form template from scratch without knowing to Give core. Most developers will copy the template which they want to overwrite into the there custom form template.

Recurring and Fee recovery addon both does not have template functions to output something on receipt but still, it is a good idea to group them in a specific class to make them easily accessible.

Reply:
I made a few changes related to this: https://github.com/impress-org/givewp/compare/ccc32725cbfe...7fe379dce00b